### DISCUSSION TITLE
第5节课作业(19班)

### DISCUSSION THREAD

**Discussion by LYMDLUT** [InternLM/Tutorial#331] [作业]
第5节课作业(19班)

**Comment by hu-qi** [comment]
- 基础作业：

使用 LMDeploy 以本地对话、网页Gradio、API服务中的一种方式部署 InternLM-Chat-7B 模型，生成 300 字的小故事（需截图）

本地对话：
![image](https://github.com/InternLM/tutorial/assets/17986122/0705e01c-8fba-4c25-8699-f6162d5523e4)


API服务：
![image](https://github.com/InternLM/tutorial/assets/17986122/b0b0cae7-2db2-4ee6-b9b9-4c590bfa9fc1)

网页Gradio:
![image](https://github.com/InternLM/tutorial/assets/17986122/0dcea06e-193a-423f-b986-54c0037c8555)



~- 进阶作业（可选做）~

~将第四节课训练自我认知小助手模型使用 LMDeploy 量化部署到 OpenXLab 平台。~
~对internlm-chat-7b模型进行量化，并同时使用KV Cache量化，使用量化后的模型完成API服务的部署，分别对比模型量化前后（将 bs设置为 1 和 max len 设置为512）和 KV Cache 量化前后（将 bs设置为 8 和 max len 设置为2048）的显存大小。~
~在自己的任务数据集上任取若干条进行Benchmark测试，测试方向包括：~
~（1）TurboMind推理+Python代码集成~
~（2）在（1）的基础上采用W4A16量化~
~（3）在（1）的基础上开启KV Cache量化~
~（4）在（2）的基础上开启KV Cache量化~
~（5）使用Huggingface推理~

**Comment by romanwin28** [comment]
基础作业：
本地：
<img width="922" alt="微信截图_20240116152949" src="https://github.com/InternLM/tutorial/assets/146464537/535c7bb0-dadd-49ac-97fe-4d5f826ef81e">

**Comment by Jiangzzzzl** [comment]
<img width="1274" alt="48a8950be1ac1e8e8167f0244151f1a" src="https://github.com/InternLM/tutorial/assets/150671792/854dd8f4-805c-40b1-b861-7d5faa5c7553">
<img width="1274" alt="c3deed20248d28da91e7b72721234f4" src="https://github.com/InternLM/tutorial/assets/150671792/68fd57e1-067e-4372-8860-e4544947ea16">

**Comment by jts250** [comment]
https://github.com/jts250/homework_llm/blob/main/homework5.pdf

**Comment by melonwine** [comment]
https://github.com/melonwine/InternLMPlay/blob/master/05/asgmt05.md

**Comment by Sixdes** [comment]
https://github.com/Sixdes/puyu-course/tree/main/homework0117

**Comment by AllYoung** [comment]
https://blog.csdn.net/sunshine_youngforyou/article/details/135682182

**Comment by mkdirmushroom** [comment]
[第5节课作业](https://github.com/mkdirmushroom/note/blob/main/InternLM/tutorial4_assignment.md)

**Comment by Hamasuta666** [comment]
第五节课笔记（作业在文末）：https://zhuanlan.zhihu.com/p/680990420
